runtime/js/fs_node.js

```diff
@@ -320,128 +337,142 @@ function fs_node_stats_from_js(js_stats, large) {
+class MlNodeDevice {}
 
 //Provides: MlNodeFd
 //Requires: MlFile, caml_uint8_array_of_string, caml_uint8_array_of_bytes, caml_bytes_set, caml_raise_sys_error
 //Requires: caml_raise_nodejs_error, caml_raise_system_error, fs_node_stats_from_js
-function MlNodeFd(fd, flags) {
-  this.fs = require("node:fs");
-  this.fd = fd;
-  this.flags = flags;
-  var stats = this.fs.fstatSync(fd);
-  flags.noSeek =
-    stats.isCharacterDevice() || stats.isFIFO() || stats.isSocket();
-  this.offset = this.flags.append ? stats.size : 0;
-  this.seeked = false;
-}
-MlNodeFd.prototype = new MlFile();
-MlNodeFd.prototype.constructor = MlNodeFd;
-
-MlNodeFd.prototype.truncate = function (len, raise_unix) {
-  try {
-    this.fs.ftruncateSync(this.fd, len | 0);
-    if (this.offset > len) this.offset = len;
-  } catch (err) {
-    caml_raise_nodejs_error(err, raise_unix);
+class MlNodeFd extends MlFile {
+  constructor(fd, flags) {
+    super();
+    this.fs = require("node:fs");
+    this.fd = fd;
+    this.flags = flags;
+    var stats = this.fs.fstatSync(fd);
+    flags.noSeek =
+      stats.isCharacterDevice() || stats.isFIFO() || stats.isSocket();
+    this.offset = this.flags.append ? stats.size : 0;
+    this.seeked = false;
   }
-};
-MlNodeFd.prototype.length = function () {
-  try {
-    return this.fs.fstatSync(this.fd).size;
-  } catch (err) {
-    caml_raise_sys_error(err.toString());
+
+  truncate(len, raise_unix) {
+    try {
+      this.fs.ftruncateSync(this.fd, len | 0);
+      if (this.offset > len) this.offset = len;
+    } catch (err) {
+      caml_raise_nodejs_error(err, raise_unix);
+    }
   }
-};
-MlNodeFd.prototype.write = function (buf, buf_offset, len, raise_unix) {
-  try {
-    if (this.flags.noSeek || !this.seeked)
-      var written = this.fs.writeSync(this.fd, buf, buf_offset, len);
-    else
-      var written = this.fs.writeSync(
-        this.fd,
-        buf,
-        buf_offset,
-        len,
-        this.offset,
-      );
-    this.offset += written;
-  } catch (err) {
-    caml_raise_nodejs_error(err, raise_unix);
+
+  length() {
+    try {
+      return this.fs.fstatSync(this.fd).size;
+    } catch (err) {
+      caml_raise_sys_error(err.toString());
+    }
   }
-  return written;
-};
-MlNodeFd.prototype.read = function (a, buf_offset, len, raise_unix) {
-  try {
-    if (this.flags.noSeek || !this.seeked)
-      var read = this.fs.readSync(this.fd, a, buf_offset, len);
-    else var read = this.fs.readSync(this.fd, a, buf_offset, len, this.offset);
-    this.offset += read;
-    return read;
-  } catch (err) {
-    caml_raise_nodejs_error(err, raise_unix);
+
+  write(buf, buf_offset, len, raise_unix) {
+    try {
+      var written;
+      if (this.flags.noSeek || !this.seeked) {
+        written = this.fs.writeSync(this.fd, buf, buf_offset, len);
+      } else {
+        written = this.fs.writeSync(this.fd, buf, buf_offset, len, this.offset);
+      }
+      this.offset += written;
+      return written;
```

Comment:
Please try to avoid mixing unrelated changes like this one (Especially in the same commit). It makes me spend much more time than I would like on review, trying to check equivalence in a sea of changes.

Reply:
Oops, I thought I put this back in the last revision